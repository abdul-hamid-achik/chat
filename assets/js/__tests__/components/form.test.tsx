import React from "react"
import { render, act, waitFor } from "@testing-library/react"
import Form from "~/components/form"

const submitMock = jest.fn()

describe("`<Form />`", () => {
    let component
    beforeEach(async () => {
        act(() => {
            component = render(<Form action="/url" method="get" submit={submitMock} />)
        })

        await waitFor(() => component)
    })
    it("renders correctly and matches snapshot", () => {
        expect(component).toMatchSnapshot()
    })
})