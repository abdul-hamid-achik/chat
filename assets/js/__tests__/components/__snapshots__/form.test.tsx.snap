// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`<Form />\` renders correctly and matches snapshot 1`] = `
<form
  action="/url"
  method="get"
  onSubmit={[MockFunction]}
/>
`;
